posthog/models/feature_flag.py

```diff
@@ -164,137 +165,159 @@ def group_type_index_to_name(self) -> Dict[GroupTypeIndex, GroupTypeName]:
             value_min = value_max
         return lookup_table
 
     @cached_property
-    def query_conditions(self) -> List[List[bool]]:
-        if self.feature_flag.aggregation_group_type_index is None:
-            query: QuerySet = Person.objects.filter(
-                team_id=self.feature_flag.team_id,
-                persondistinctid__distinct_id=self.distinct_id,
-                persondistinctid__team_id=self.feature_flag.team_id,
-            )
-        else:
-            query = Group.objects.filter(
-                team_id=self.feature_flag.team_id,
-                group_type_index=self.feature_flag.aggregation_group_type_index,
-                group_key=self.hashed_identifier,
-            )
-
-        fields = []
-        for index, condition in enumerate(self.feature_flag.conditions):
-            key = f"condition_{index}"
-
-            if len(condition.get("properties", {})) > 0:
-                # Feature Flags don't support OR filtering yet
-                expr: Any = properties_to_Q(
-                    Filter(data=condition).property_groups.flat, team_id=self.feature_flag.team_id, is_direct_query=True
-                )
-            else:
-                expr = RawSQL("true", [])
-
-            query = query.annotate(**{key: ExpressionWrapper(expr, output_field=BooleanField())})
-            fields.append(key)
-
-        return list(query.values_list(*fields))
-
-    @cached_property
-    def hashed_identifier(self) -> Optional[str]:
+    def query_conditions(self) -> Dict[str, bool]:
+        team_id = self.feature_flags[0].team_id
+        person_query: QuerySet = Person.objects.filter(
+            team_id=team_id, persondistinctid__distinct_id=self.distinct_id, persondistinctid__team_id=team_id,
+        )
+        group_query: QuerySet = Group.objects.filter(team_id=team_id,)
+        person_fields = []
+        group_fields = []
+
+        for feature_flag in self.feature_flags:
+            for index, condition in enumerate(feature_flag.conditions):
+                key = f"flag_{feature_flag.pk}_condition_{index}"
+                expr: Any = None
+                if len(condition.get("properties", {})) > 0:
+                    # Feature Flags don't support OR filtering yet
+                    expr = properties_to_Q(
+                        Filter(data=condition).property_groups.flat, team_id=team_id, is_direct_query=True
+                    )
+
+                if feature_flag.aggregation_group_type_index is None:
+                    person_query = person_query.annotate(
+                        **{key: ExpressionWrapper(expr if expr else RawSQL("true", []), output_field=BooleanField())}
+                    )
+                    person_fields.append(key)
+                else:
+                    group_filter = Q(
+                        group_type_index=self.feature_flags[0].aggregation_group_type_index,
```

Comment:
oh, I'm being dumb, I guess this was supposed to be like so:
```suggestion
                        group_type_index=feature_flag.aggregation_group_type_index,
```

Reply:
Yep, good catch!